### PR TITLE
MRG: Fix BrainVision date error

### LINKED ISSUE #6054: brainvision _str_to_meas_date breaks with incorrect date
if the original brainvision file lacks date information and we do segmentation on it, the offset of "New Segment" is added to "000000000000000000" so we end with something like '00000000000304125000' which passes sanity check in _str_to_meas_date, but it is stil invalid:

`def _str_to_meas_date(date_str):
    date_str = date_str.strip()

    if date_str in ['0', '00000000000000000000']:
        return None`

the solution is to add additional check:

```
    try:
        meas_date = datetime.strptime(date_str, '%Y%m%d%H%M%S%f')
    except ValueError:
        return None
```

### PR DESCRIPTION
**Pull request by sappelhoff**
Fixes #6054 

Changes:

- Removing a superfluous  conditional check that is already handled shortly afterwards 
- Introducing a test case that reproduces the error described by @petergrey75 in #6054 
- Inserting a try-except clause in `_str_to_meas_date` to fix the error

### CODE REVIEW

File: mne/io/brainvision/tests/test_brainvision.py [hunk 1]
```diff
@@ -91,6 +94,12 @@ def test_vmrk_meas_date():
     assert raw.info['meas_date'] is None
     assert 'unspecified' in repr(raw.info)
 
+    # Test files with faulty dates introduced by segmenting a file without
+    # date information. Should not rais a strptime ValueError
```
Comment (cbrnr) [inline mne/io/brainvision/tests/test_brainvision.py]: rais -> raise

**Comment by agramfort** [comment]
@massich can you have a look?

**Comment by larsoner** [comment]
@cbrnr feel free to merge if you are happy

**Comment by larsoner** [comment]
@massich it's okay to set `info['meas_date']` to `None` nowadays, right?

**Comment by jona-sassenhagen** [comment]
Travis has started, I will merge if it comes back green.

**Comment by larsoner** [comment]
Travis is hours behind us :(

**Comment by larsoner** [comment]
(We can probably trust Azure for now so we don't get a backlog, and I'll fix Travis in the morning if it's broken)

**Comment by larsoner** [comment]
Thanks @sappelhoff

**Comment by jona-sassenhagen** [comment]
>(We can probably trust Azure for now so we don't get a backlog

Narrator: They couldn't.

**Pull request merged**